conda_smithy/templates/build_steps.sh.tmpl

```diff
@@ -29,11 +31,16 @@ CONDARC
 {%- if build_with_mambabuild %}
 GET_BOA=boa
 BUILD_CMD=mambabuild
+CONDA_INSTALL_CMD=mamba
 {%- else %}
 BUILD_CMD=build
+CONDA_INSTALL_CMD=conda
 {%- endif %}
 
-conda install --yes --quiet "{{ remote_ci_setup }}" conda-build pip ${GET_BOA:-} -c conda-forge
+{%- if build_with_mambabuild %}
+${CONDA_INSTALL_CMD} update --all --yes --quiet
```

Comment:
This is going to invalidate the pins like `python=3.9` we have added in the docker image.

Reply:
I think that is wrong. Here is mamba on my local machine:

```
$ mamba update --all --yes

Looking for: ['libcurl', 'blinker', 'pexpect', 'gettext', 'libgcrypt', 'jsonschema', 'argon2-cffi', 'pyyaml', 'tomlkit', 'matplotlib-base', 'certifi', 'mamba', 'py-lief', 'nbformat', 'six', 'pygments', 'gitpython', 'keyring', 'liblief', 'pyzmq', 'toml', 'jq', 'qtconsole', 'notebook', 'brotli', 'iniconfig', 'ncurses', 'requests', 'zeromq', 'libmamba', 'libgpg-error', 'jupyter_console', 'cloudpickle', 'pynamodb', 'bleach', 'botocore', 'parameterized', 'send2trash', 'mccabe', 'libuv', 'make', 'rfc3986', 'pyqt-impl', 'lcms2', 'filelock', 'cycler', 'attrs', 'jupyterlab_pygments', 'pkg-config', 'testpath', 'reproc-cpp', 'reproc', 'popt', 'libglib', 'tornado', 'glib-tools', 'backcall', 'cffi', 'pandoc', 'ruamel.yaml.jinja2', 'pyparsing', 'libcxx', 'glob2', 'python-dateutil', 'zlib', 'boa', 'kiwisolver', 'pyopenssl', 'jwcrypto', 'conda-forge-pinning', 'liblapack', 'pysocks', 'coverage', 'libpq', 'uritemplate', 'lerc', 'nest-asyncio', 'more-itertools', 'simplejson', 'appdirs', 'qt', 'smmap', 'pyrsistent', 'libwebp-base', 'importlib_metadata', 'pcre', 'libedit', 'conda-build', 'ptyprocess', 'commonmark', 'pyasn1', 'python-libarchive-c', 'packaging', 'glib', 'numpy', 'jupyter_highlight_selected_word', 'prometheus_client', 'conda-package-handling', 'github3.py', 'pycosat', 'pandocfilters', 'python_abi', 'setproctitle', 'libiconv', 'ipywidgets', 'pyqtwebengine', 'nb_conda_kernels', 'pyjwt', 'munkres', 'backports', 'mysql-common', 'platformdirs', 'olefile', 'cfitsio', 'wheel', 'webencodings', 'xz', 'charset-normalizer', 'chardet', 'jinja2', 'joblib', 'git', 'brotli-bin', 'pip', 'perl', 'appnope', 'setuptools-scm', 'zipp', 'curl', 'license-expression', 'brotlipy', 'colorama', 'gitdb', 'nbclient', 'psutil', 'toolz', 'requests-toolbelt', 'rpm-tools', 'ipython', 'scrypt', 'tk', 'jbig', 'boto3', 'ruamel_yaml', 'jupyter_contrib_nbextensions', 'archspec', 'requests-mock', 'ipykernel', 'libtiff', 'click', 'libgfortran', 'tomli', 'jpeg', 'py', 'pathspec', 'pycrypto', 'sqlite', 'libbrotlidec', 'krb5', 'soupsieve', 'watchgod', 'oauthlib', 'prompt-toolkit', 'hexalattice', 'conda-suggest-conda-forge', 'readme_renderer', 'libcblas', 'widgetsnbextension', 'black', 'docutils', 'libdeflate', 'icu', 'llvm-openmp', 'libffi', 'requests-oauthlib', 'jedi', 'defusedxml', 'typing-extensions', 'uritemplate.py', 'libnghttp2', 'jupyter', 'libdb', 'prompt_toolkit', 'conda-smithy', 'rich', 'libsolv', 'jupyter_latex_envs', 'mysql-libs', 'zstd', 'mock', 'nspr', 'libopenblas', 'file', 'libarchive', 'jupyter_contrib_core', 'pyqtchart', 'frozendict', 'jupyter_nbextensions_configurator', 'tar', 'typed-ast', 'libllvm11', 'm2r', 'importlib_resources', 'yaml', 'jupyter_core', 'entrypoints', 'readline', 'openssl', 'ndg-httpsclient', 'libxml2', 'twine', 'libbrotlicommon', 'anaconda-client', 'loky', 'markupsafe', 'boolean.py', 'pluggy', 'libzlib', 'bzip2', 'libmagic', 'c-ares', 'jupyter_client', 'beautifulsoup4', 'json5', 'lazyasd', 'python', 'cmarkgfm', 'pkginfo', 'ripgrep', 'nodejs', 's3transfer', 'et_xmlfile', 'pytest-cov', 'argcomplete', 'libxslt', 'unicodedata2', 'pyqt5-sip', 'libbrotlienc', 'ruamel.yaml', 'setuptools', 'libssh2', 'wcwidth', 'yaml-cpp', 'matplotlib', 'jupyterlab_widgets', 'libsodium', 'jdcal', 'clyent', 'pygithub', 'pcre2', 'isodate', 'codecov', 'lz4-c', 'ca-certificates', 'pillow', 'python.app', 'importlib-metadata', 'idna', 'async_generator', 'pytest', 'mypy_extensions', 'tqdm', 'cachetools', 'pycodestyle', 'wurlitzer', 'qtpy', 'nss', 'jmespath', 'setuptools_scm', 'libclang', 'pyflakes', 'backoff', 'conda', 'pynacl', 'libpng', 'debugpy', 'openpyxl', 'dbus', 'pytz', 'pyqt', 'ldid', 'fonttools', 'lxml', 'nbconvert', 'openjpeg', 'regex', 'msrest', 'terminado', 'ipython_genutils', 'dataclasses', 'pickleshare', 'inotify_simple', 'lzo', 'urllib3', 'vsts-python-api', 'typing_extensions', 'libblas', 'pybind11-abi', 'matplotlib-inline', 'wrapt', 'decorator', 'mistune', 'lua', 'deprecated', 'backports.functools_lru_cache', 'pycparser', 'parso', 'libgfortran5', 'tzdata', 'expat', 'libev', 'future', 'traitlets', 'libmambapy', 'oniguruma', 'ruamel.yaml.clib', 'conda-suggest', 'setuptools_scm_git_archive', 'freetype', 'cryptography', 'flake8']

pkgs/main/noarch         [====================] (00m:00s) No change
pkgs/r/osx-64            [====================] (00m:00s) No change
pkgs/main/osx-64         [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change
conda-forge/noarch       [====================] (00m:02s) Done
conda-forge/osx-64       [====================] (00m:05s) Done

Pinned packages:
  - python 3.8.*


Transaction

  Prefix: /Users/beckermr/miniconda3

  Updating specs:

   - libcurl
   - blinker
   - pexpect
   - gettext
   - libgcrypt
   - jsonschema
   - argon2-cffi
   - pyyaml
   - tomlkit
   - matplotlib-base
   - certifi
   - mamba
   - py-lief
   - nbformat
   - six
   - pygments
   - gitpython
   - keyring
   - liblief
   - pyzmq
   - toml
   - jq
   - qtconsole
   - notebook
   - brotli
   - iniconfig
   - ncurses
   - requests
   - zeromq
   - libmamba
   - libgpg-error
   - jupyter_console
   - cloudpickle
   - pynamodb
   - bleach
   - botocore
   - parameterized
   - send2trash
   - mccabe
   - libuv
   - make
   - rfc3986
   - pyqt-impl
   - lcms2
   - filelock
   - cycler
   - attrs
   - jupyterlab_pygments
   - pkg-config
   - testpath
   - reproc-cpp
   - reproc
   - popt
   - libglib
   - tornado
   - glib-tools
   - backcall
   - cffi
   - pandoc
   - ruamel.yaml.jinja2
   - pyparsing
   - libcxx
   - glob2
   - python-dateutil
   - zlib
   - boa
   - kiwisolver
   - pyopenssl
   - jwcrypto
   - conda-forge-pinning
   - liblapack
   - pysocks
   - coverage
   - libpq
   - uritemplate
   - lerc
   - nest-asyncio
   - more-itertools
   - simplejson
   - appdirs
   - qt
   - smmap
   - pyrsistent
   - libwebp-base
   - importlib_metadata
   - pcre
   - libedit
   - conda-build
   - ptyprocess
   - commonmark
   - pyasn1
   - python-libarchive-c
   - packaging
   - glib
   - numpy
   - jupyter_highlight_selected_word
   - prometheus_client
   - conda-package-handling
   - github3.py
   - pycosat
   - pandocfilters
   - python_abi
   - setproctitle
   - libiconv
   - ipywidgets
   - pyqtwebengine
   - nb_conda_kernels
   - pyjwt
   - munkres
   - backports
   - mysql-common
   - platformdirs
   - olefile
   - cfitsio
   - wheel
   - webencodings
   - xz
   - charset-normalizer
   - chardet
   - jinja2
   - joblib
   - git
   - brotli-bin
   - pip
   - perl
   - appnope
   - setuptools-scm
   - zipp
   - curl
   - license-expression
   - brotlipy
   - colorama
   - gitdb
   - nbclient
   - psutil
   - toolz
   - requests-toolbelt
   - rpm-tools
   - ipython
   - scrypt
   - tk
   - jbig
   - boto3
   - ruamel_yaml
   - jupyter_contrib_nbextensions
   - archspec
   - requests-mock
   - ipykernel
   - libtiff
   - click
   - libgfortran
   - tomli
   - jpeg
   - py
   - pathspec
   - pycrypto
   - sqlite
   - libbrotlidec
   - krb5
   - soupsieve
   - watchgod
   - oauthlib
   - prompt-toolkit
   - hexalattice
   - conda-suggest-conda-forge
   - readme_renderer
   - libcblas
   - widgetsnbextension
   - black
   - docutils
   - libdeflate
   - icu
   - llvm-openmp
   - libffi
   - requests-oauthlib
   - jedi
   - defusedxml
   - typing-extensions
   - uritemplate.py
   - libnghttp2
   - jupyter
   - libdb
   - prompt_toolkit
   - conda-smithy
   - rich
   - libsolv
   - jupyter_latex_envs
   - mysql-libs
   - zstd
   - mock
   - nspr
   - libopenblas
   - file
   - libarchive
   - jupyter_contrib_core
   - pyqtchart
   - frozendict
   - jupyter_nbextensions_configurator
   - tar
   - typed-ast
   - libllvm11
   - m2r
   - importlib_resources
   - yaml
   - jupyter_core
   - entrypoints
   - readline
   - openssl
   - ndg-httpsclient
   - libxml2
   - twine
   - libbrotlicommon
   - anaconda-client
   - loky
   - markupsafe
   - boolean.py
   - pluggy
   - libzlib
   - bzip2
   - libmagic
   - c-ares
   - jupyter_client
   - beautifulsoup4
   - json5
   - lazyasd
   - python
   - cmarkgfm
   - pkginfo
   - ripgrep
   - nodejs
   - s3transfer
   - et_xmlfile
   - pytest-cov
   - argcomplete
   - libxslt
   - unicodedata2
   - pyqt5-sip
   - libbrotlienc
   - ruamel.yaml
   - setuptools
   - libssh2
   - wcwidth
   - yaml-cpp
   - matplotlib
   - jupyterlab_widgets
   - libsodium
   - jdcal
   - clyent
   - pygithub
   - pcre2
   - isodate
   - codecov
   - lz4-c
   - ca-certificates
   - pillow
   - python.app
   - importlib-metadata
   - idna
   - async_generator
   - pytest
   - mypy_extensions
   - tqdm
   - cachetools
   - pycodestyle
   - wurlitzer
   - qtpy
   - nss
   - jmespath
   - setuptools_scm
   - libclang
   - pyflakes
   - backoff
   - conda
   - pynacl
   - libpng
   - debugpy
   - openpyxl
   - dbus
   - pytz
   - pyqt
   - ldid
   - fonttools
   - lxml
   - nbconvert
   - openjpeg
   - regex
   - msrest
   - terminado
   - ipython_genutils
   - dataclasses
   - pickleshare
   - inotify_simple
   - lzo
   - urllib3
   - vsts-python-api
   - typing_extensions
   - libblas
   - pybind11-abi
   - matplotlib-inline
   - wrapt
   - decorator
   - mistune
   - lua
   - deprecated
   - backports.functools_lru_cache
   - pycparser
   - parso
   - libgfortran5
   - tzdata
   - expat
   - libev
   - future
   - traitlets
   - libmambapy
   - oniguruma
   - ruamel.yaml.clib
   - conda-suggest
   - setuptools_scm_git_archive
   - freetype
   - cryptography
   - flake8
   - ca-certificates
   - certifi
   - openssl


  Package                            Version  Build         Channel                  Size
───────────────────────────────────────────────────────────────────────────────────────────
  Upgrade:
───────────────────────────────────────────────────────────────────────────────────────────

  - boto3                            1.20.20  pyhd8ed1ab_0  installed                    
  + boto3                            1.20.21  pyhd8ed1ab_0  conda-forge/noarch      71 KB
  - botocore                         1.23.20  pyhd8ed1ab_0  installed                    
  + botocore                         1.23.21  pyhd8ed1ab_0  conda-forge/noarch       5 MB
  - conda-forge-pinning  2021.12.05.20.40.01  hd8ed1ab_0    installed                    
  + conda-forge-pinning  2021.12.07.13.39.21  hd8ed1ab_0    conda-forge/noarch      18 KB
  - glib                              2.70.1  hcf210ce_0    installed                    
  + glib                              2.70.2  hcf210ce_0    conda-forge/osx-64     425 KB
  - glib-tools                        2.70.1  hcf210ce_0    installed                    
  + glib-tools                        2.70.2  hcf210ce_0    conda-forge/osx-64      93 KB
  - libglib                           2.70.1  hf1fb8c0_0    installed                    
  + libglib                           2.70.2  hf1fb8c0_0    conda-forge/osx-64       3 MB

  Summary:

  Upgrade: 6 packages

  Total download: 9 MB

───────────────────────────────────────────────────────────────────────────────────────────

Finished glib-tools                           (00m:00s)              93 KB    334 KB/s
Finished conda-forge-pinning                  (00m:00s)              18 KB     63 KB/s
Finished glib                                 (00m:00s)             425 KB      1 MB/s
Finished boto3                                (00m:00s)              71 KB    201 KB/s
Finished libglib                              (00m:00s)               3 MB      6 MB/s
Finished botocore                             (00m:00s)               5 MB      8 MB/s
Downloading  [====================================================================================================] (00m:05s)   13.55 MB/s
Extracting   [====================================================================================================] (00m:05s)        6 / 6
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

Notice it reports python as pinned.

Reply:
Can you open a PR in a feedstock with these changes?

Reply:
Yes!

Reply:
See here: https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock/pull/321